### PR TITLE
Support formatting non quantitative encodings

### LINKED ISSUE #4138: legend and axis format property has no effect for ordinal encodings
Illustrated by the following spec:

When you specify ``format`` in a vega legend, the generated vega spec does not have any ``format`` property present:

```json
{
  "data": {
    "values": [
      {"x": 0}, {"x": 1}, {"x": 2}, {"x": 3}, {"x": 4}, {"x": 5}, {"x": 6}
    ]
  },
  "transform": [{"calculate": "sin(datum.x)", "as": "y"}],
  "mark": "point",
  "encoding": {
    "color": {
      "type": "ordinal",
      "field": "y",
      "legend": {"format": ".2f"},
      "scale": {"scheme": "reds"}
    },
    "x": {"type": "quantitative", "field": "x"},
    "y": {"type": "quantitative", "field": "y"}
  }
}
```

[Open in Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4yghSA3WAbArgKYDOKYA2uBhMJAB5kAMAvgDRXW0OoCMbHGLmQBM-ajXpkAzGPFDUAFlmdJqAKzLBqsADZmAgLpVZkeACdYAO2IAzAPZmAtmXK0Axnjf5cCQmUjEAJaWABRw8PiOAHR0AJSQrFCwpKiQAJ6QzAbsmI6wZgDW-gAOdsHwCVSQhJZudtDBAOZk6BiQdbgOLQKY8GnFfqkODZZ4leJQNoGEuND+GTnikLiEjTVzqLT2Tgj+UcI2mYvUAR4rLacAFoSOg1BmhNCk+tSa2rR9A-4AjvhW8IFEACsH5EpApjMNlAGG8MptTP07pBfv9AQhAiCEpNprN5pljCBmEA)
visualization 43

### PR DESCRIPTION
**Pull request by domoritz**
Fixes #4138

### CODE REVIEW

File: src/compile/common.ts [hunk 1]
```diff
@@ -127,19 +127,19 @@ export function formatSignalRef(
...
-    // add number format for quantitative type only
+  if (isTimeFieldDef(fieldDef)) {
+    return undefined;
+  }
 
-    // Specified format in axis/legend has higher precedence than fieldDef.format
-    if (specifiedFormat) {
-      return specifiedFormat;
-    }
+  // Specified format in axis/legend has higher precedence than fieldDef.format
```
Comment (kanitw) [inline src/compile/common.ts]: Isnt the comment outdated?  There is fieldDef.format anymore?

Comment (domoritz) [reply src/compile/common.ts]: Then it is

**Comment by kanitw** [comment]
Send follow up update?

File: src/compile/common.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/common.ts]: We have ti for text: https://vega.github.io/vega-lite/docs/format.html

**Pull request merged**